### PR TITLE
fix: ensure that error is not thrown on undefined block

### PR DESCRIPTION
**Pull request by stalniy**
## Why

Screenshot 2025-09-08 at 12 51 52

## Summary by CodeRabbit

- Bug Fixes
  - Improved reliability of the bid countdown: better initialization and display when block data is missing or delayed.
  - Prevents premature or missing countdown updates.

- Improvements
  - Zero-bid trial warning now evaluated more predictably shortly after load (reduces repeated checks).

- Chores
  - Switched underlying HTTP client for more stable network behavior.

- Tests
  - Expanded tests for block-edge cases and trial scenarios; updated test fixtures to cover null/variant block data.

### CODE REVIEW

**Review by baktun14 — DISMISSED** [review]

File: apps/deploy-web/src/components/new-deployment/BidCountdownTimer.tsx [hunk 1]
```diff
@@ -17,20 +17,18 @@ const time = 5 * 60;
...
       // add 20 seconds for the delay between deployment creation and bid creation
       const diff = Math.max(0, time - differenceInSeconds(now, date) + 20);
       setTimeLeft(diff);
       setIsTimerInit(true);
-    }
-  });
-
-  useEffect(() => {
-    getBlock();
+    });
```
[the remark this answers is not in the document]
Comment (stalniy) [reply apps/deploy-web/src/components/new-deployment/BidCountdownTimer.tsx]: but onSuccess just not available in useBlock options in the latest react-query

**Review by baktun14 — APPROVED** [review]
